tree/tree/src/TBranchElement.cxx

```diff
@@ -2244,6 +2250,59 @@ TClass* TBranchElement::GetCurrentClass()
    return cl;
 }
 
+void TBranchElement::InitializeBranchLists()
+{
+   const auto nbranches = fBranches.GetEntriesFast();
+
+   // If the branch holds a class featuring schema rules, we cannot parallelise
+   // since a certain order is to be imposed
+   if(auto heldClass = GetClass()) { // perhaps always true
+      if (heldClass->GetSchemaRules()) {
```

Comment:
Maybe make a note that this is a super-set of the case where we really need this.  Technically only the branch that are source or target of an 'active' (See for example how fReadActionSequence is set).

Reply:
Will do. We need to start with something solid and further increase the parallelism afterwards.